### PR TITLE
Add test case about format "a % b"

### PR DESCRIPTION
**Pull request by hicqu**
There is a bug about format `a % b` caused by go interprets `%` as a special things. The PR fixes this and add a test case.

### CODE REVIEW

File: ast/format_test.go [hunk 1]
```diff
@@ -65,6 +65,9 @@ func (ts *testAstFormatSuite) TestAstFormat(c *C) {
 		{`"abc" not regexp '.*bc?'`, `"abc" NOT REGEXP ".*bc?"`},
 		{`-  4`, `-4`},
 		{`- ( - 4 ) `, `-(-4)`},
+		{`a%b`, "`a` % `b`"},
```
Comment (tiancaiamao) [inline ast/format_test.go]: I test use the old parser, and this test case passes @hicqu

File: ast/expressions.go [hunk 2]
```diff
@@ -137,7 +138,13 @@ func (n *BinaryOperationExpr) Format(w io.Writer) {
 	fmt.Fprint(w, " ")
 	n.Op.Format(w)
```
Comment (tiancaiamao) [inline ast/expressions.go]: WriteString will not consider the % escape, it must be something in fmt.Sprint that %% => %

**Comment by hicqu** [comment]
The bug is already fixed, so just keep test cases. PTAL @tiancaiamao .

**Comment by tiancaiamao** [comment]
LGTM
b.t.w, this should already been covered here https://github.com/pingcap/parser/blob/master/opcode/opcode_test.go#L33

**Comment by tiancaiamao** [comment]
PTAL  @XuHuaiyu

File: ast/format_test.go [hunk 3]
```diff
@@ -65,6 +65,9 @@ func (ts *testAstFormatSuite) TestAstFormat(c *C) {
...
+		{`a%b+6`, "`a` % `b` + 6"},
```
Comment (ngaut) [inline ast/format_test.go]: Use ``` instead of %

File: ast/format_test.go [hunk 4]
```diff
@@ -65,6 +65,9 @@ func (ts *testAstFormatSuite) TestAstFormat(c *C) {
...
+		{`a%(b+6)`, "`a` % (`b` + 6)"},
```
Comment (ngaut) [inline ast/format_test.go]: Ditto

File: ast/format_test.go (hunk 3 above)
Comment (hicqu) [reply ast/format_test.go]: Do you mean put "`a` % `b`" in ``? I have tried but I meet another problem: `a` and `b` are already surround by ``, so we can't use `` to surround them.

**Pull request merged**